namerd/storage/k8s/src/main/scala/io/buoyant/namerd/storage/experimental/kubernetes/DtabWatch.scala

```diff
@@ -0,0 +1,23 @@
+package io.buoyant.namerd.storage.experimental.kubernetes
+
+import com.fasterxml.jackson.annotation.{JsonProperty, JsonSubTypes, JsonTypeInfo}
+import io.buoyant.k8s.{Status, Watch}
+
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.PROPERTY, property = "type")
+@JsonSubTypes(Array(
+  new JsonSubTypes.Type(value = classOf[DtabAdded], name = "ADDED"),
+  new JsonSubTypes.Type(value = classOf[DtabModified], name = "MODIFIED"),
+  new JsonSubTypes.Type(value = classOf[DtabDeleted], name = "DELETED"),
+  new JsonSubTypes.Type(value = classOf[DtabError], name = "ERROR")
+))
+sealed trait DtabWatch extends Watch[Dtab]
+
+// NOTE: These are not (as would be usual practice) inside a DtabWatch companion object,
+// because doing so leads to intermittent compilation failures from
+// https://issues.scala-lang.org/browse/SI-7551.
+case class DtabAdded(`object`: Dtab) extends DtabWatch with Watch.Added[Dtab]
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [`object` is a reserved keyword don't use it as a name](https://www.codacy.com/app/ver/linkerd/file/2415518825/issues/source?bid=3236810&fileBranchId=3236828#l18)